haystack/utils/openai_utils.py

```diff
@@ -65,34 +65,38 @@ def _openai_text_completion_tokenization_details(model_name: str):
     :param model_name: Name of the OpenAI model.
     """
     tokenizer_name = "gpt2"
-    max_tokens_limit = 2049  # Based on this ref: https://platform.openai.com/docs/models/gpt-3
+    max_tokens_limit = 4096  # It is the minimum max_tokens_limit value based on this ref: https://platform.openai.com/docs/models/overview
+
     try:
         model_tokenizer = tiktoken.encoding_name_for_model(model_name)
     except KeyError:
         model_tokenizer = None
 
     if model_tokenizer:
-        # Based on OpenAI models page, 'davinci' considers have 2049 tokens,
-        ## therefore, it is better to add `text-davinci` instead to the condition.
-        ## Ref: https://platform.openai.com/docs/models/gpt-3-5
-        ##      https://platform.openai.com/docs/models/gpt-3
-        if "text-davinci" in model_name:
-            max_tokens_limit = 4097
-            tokenizer_name = model_tokenizer
-        elif model_name.startswith("gpt-3.5-turbo-16k") or model_name.startswith("gpt-35-turbo-16k"):
+        # Based on OpenAI models page, the following are the max_tokens_limit values for the corresponding models
+        ## Ref: https://platform.openai.com/docs/models/overview
+        if "davinci" in model_name:
             max_tokens_limit = 16384
             tokenizer_name = model_tokenizer
-        elif model_name.startswith("gpt-3.5-turbo-1106") or model_name.startswith("gpt-35-turbo-1106"):
-            max_tokens_limit = 16384
+        elif model_name.startswith("gpt-3.5-turbo-instruct") or model_name.startswith("gpt-35-turbo-instruct"):
```

Comment:
We don't need a check for `gpt-3.5-turbo-instruct` since the max_token_limit is the same as the new default.

Reply:
I added this check since the default tokenizer is different ("cl100k_base" instead of "gpt2") as you can see throught the unit test test_openai_text_completion_tokenization_details_gpt_3_5_turbo_instruct in test_openai_utils.py

Reply:
Hmm I would have thought we would still set the right tokenizer due to these lines https://github.com/deepset-ai/haystack/blob/d3daad0d875e7610beb83e5c1a0f8f3167de22d4/haystack/utils/openai_utils.py#L105-L106 at the end of the if statement.

Reply:
Ah ok, I got it! I suggest that the default tokenizer should be set to "cl100k_base" since basically all openai models now support this tokenizer. There are only few exceptions left with the "gpt2" tokenizer. As you can see [here](https://github.com/openai/tiktoken/blob/main/tiktoken/model.py) most of the "gpt2" models are now deprecated

Reply:
That's a good point, I think we had it as `gpt2` before because tiktoken used to be an optional dependency. But it's required now so I think updating to `cl100k_base` as default makes sense to me.